submissions/YuraZagor/tjsw/index.js

```diff
@@ -0,0 +1,166 @@
+const manNamesArr = ['Vasil', 'Yurij', 'Zahar', 'Taras', 'Olexiy', 'Oleg', 'Mykola', 'Sashko', 'Sergij', 'Bohdan'];
+const womanNamesArr = ['Hanna', 'Helena', 'Vasilisa', 'Yulya', 'Oksana', 'Olena', 'Olga', 'Myroslava', 'Svitlana', 'Bohdana'];
+const animalNamesArr = ['Joy', 'Happy', 'Lucky', 'Puffy'];
+const genderArr = ['male', 'female'];
+
+let catSay = 'meau';
+let dogSay = 'woufff';
+let humanSay = 'Glory to Ukraine!';
+
+function randomSelect(arr){
+  return arr[Math.floor(Math.random()*arr.length)]
+};
+
+let manName = randomSelect(manNamesArr);
+let womanName = randomSelect(womanNamesArr);
+let animalName = randomSelect(animalNamesArr);
+
+const inhabitants = [
+  { species: 'human', 
+    gender: 'male',
+    name: 'Vasil',
+    saying: 'Slava natsiyi',
+    friend: 'Hanna',
+    legs: 2,
+    hands: 2,
+  },
+  { species: 'human', 
+    gender: 'female',
+    name: 'Hanna',
+    saying: 'Smert vorogam',
+    friend: 'Vasil',
+    legs: 2,
+    hands: 2,
+  },];
+
+const dogsVolier = [{ 
+  species: 'Dog', 
+  gender: 'female',
+  name: 'Guffy',
+  saying: dogSay,
+  friend: randomSelect(inhabitants).name,
+  paws: 4,
+  tail: 1,
+},];
+
+class Inhabitant {  
+  constructor(species, name, saying, friend, gender, habitat = inhabitants){
+    this.species = species;
+    this.name = name;
+    this.saying = saying
+    this.friend = friend;
+    this.gender = gender;
+    this.habitat = habitat;
+  }
+  accountInhabitants(){
+    this.habitat.push(this);
+  };
+  commonProperties(){
+    const { species, name, saying, friend, gender} = this;
+    return [species, name, saying, friend, gender]
+  }
+};
+
+class Animal extends Inhabitant{
+  constructor( species, name, saying, friend, gender, paws, tail ){
+    super(species, name, saying, friend, gender);
+    this.paws = paws;
+    this.tail = tail;
+  };
+  populate(){
+    const { paws, tail} = this;
+    const speciesProperties = [ paws, tail];
+    print(this.commonProperties().concat(speciesProperties).join('; '),'div');
+    this.accountInhabitants()
+  };
+};
+
+class Dog extends Animal{
+  constructor( name, friend, gender ){
+    super('dog', name, dogSay, friend, gender, 4, 1);
+    this.habitat = dogsVolier
```

Comment:
I am confused. LoC 36-44 tell me that `dogsVolier` contain a list of dogs blueprints...
Here it is assigned as a habitat. So what `dogsVolier` is in plain language and what it contains?
What is the purpose of `habitat` property?

Reply:
habitat was introduced to try and solve the task you gave me 2 weeks ago:  _Better design would allow us to add containers like cities, communities etc - all populated with the same inhabitants (in different combinations)._